sound/soc/intel/boards/sof_sdw.c

```diff
@@ -919,6 +921,11 @@ static int create_sdw_dailink(struct snd_soc_card *card,
 
 			codecs[j].name = sof_end->codec_name;
 			codecs[j].dai_name = sof_end->dai_info->dai_name;
+			if (sof_end->dai_info->dai_type == SOC_SDW_DAI_TYPE_MIC &&
+			    mach_params->dmic_num > 0) {
+				dev_warn(dev,
+					 "Both SDW DMIC and PCH DMIC are present, if incorrect, please set kernel params snd_sof_intel_hda_generic dmic_num=0 to disable PCH DMIC\n");
```

Comment:
and what if the PCH-DMIC is the correct and the SDW DMIC is not?

Reply:
Not all SDW DMIC dai are implemented as optional and can be removed. We add `.quirk = SOC_SDW_CODEC_MIC,` in the codec_info_list[] if the SDW DMIC DAI is optional. And can remove it with snd-soc-sof-sdw quirk. But we need to add the `SOC_SDW_CODEC_MIC` quirk if the `.quirk_exclude` flag is set in the codec_info_list[] or remove the `SOC_SDW_CODEC_MIC` quirk if the `.quirk_exclude` flag is not set to remove the SDW DMIC. That's why I didn't add description for removing the SDW DMIC here.